examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,373 @@
+    # init table headers 
+    headers = ['Checkpoint']
+    headers.extend([f"{task.upper()}" for task in sorted(tasks)])
+    headers.extend(['GLUE-Large', 'GLUE-All'])
+    tb = [headers]
+
+    # fill table 
+    for ckpt in glue_metrics.get_logged_ckpts():
+        output_line = [ckpt]
+        glue_all = 0
+        glue_large = 0
+        # Per task score
+        for task in sorted(glue_metrics.task_names):
+            task_metric = 0 if not glue_metrics.get_task_metric(ckpt, task) else glue_metrics.get_task_metric(ckpt, task) 
+            output_line.append("{:.4f}".format(task_metric))
+            glue_all += task_metric
+            if task in large_tasks:
+                glue_large += task_metric
+        # GLUE Large and GLUE All
+        output_line.append("{:.4f}".format(glue_large/len(large_tasks)))
+        output_line.append("{:.4f}".format(glue_all/len(tasks)))
+        tb.append(output_line)
+    
+    print(tabulate(tb, headers='firstrow'))
+
+''' Callback function for metric collection. '''
+def log_metrics(parent_ckpt: Optional[str], ckpt_filename: str, glue_metrics: GlueState, f_metric: Future) -> None:
+    metric = f_metric.result()
+    if parent_ckpt: 
+        logged_ckpt_name = parent_ckpt
+    else:
+        logged_ckpt_name = ckpt_filename
+
+    if logged_ckpt_name not in glue_metrics.get_logged_ckpts():
+        glue_metrics.init_ckpt_dict(logged_ckpt_name)
+            
+    task = list(metric.keys())[0]
+    formatted_task = task.split('_')[1] # remove "glue_" prefix
+    for key in metric.keys(): # handle case where mnli has glue_mnli and glue_mnli_mismatched
+        for m in metric[key]:
+            if not glue_metrics.get_task_metric(logged_ckpt_name, formatted_task):
+                glue_metrics.set_task_metric(logged_ckpt_name, formatted_task, metric[key][m].item())
+            else:
+                old_val = glue_metrics.get_task_metric(logged_ckpt_name, formatted_task)
+                glue_metrics.set_task_metric(logged_ckpt_name, formatted_task, sum([old_val, metric[key][m].item()]) / 2) # automatically average scores 
+    
+''' Set up CUDA environment and process pool for given finetuning jobs. '''
+def setup_finetuning_jobs(finetune_tasks: List[str], save_ckpts: List[bool], ckpt_load_path: str, ckpt_filename: str, save_folder: str, glue_metrics: GlueState,
+         wandb_logger: Union[WandBLogger, None], save_locally: bool, bucket: Optional[str] = None, parent_ckpt: Optional[str] = None, load_ignore_keys: Optional[List[str]] = None) -> None:
+    cuda_envs = init_cuda_queue(torch.cuda.device_count())
+    num_tasks = len(finetune_tasks)
+    
+    if parent_ckpt: 
+        wandb_group_name = parent_ckpt
+    else:
+        wandb_group_name = ckpt_load_path
+
+    # finetuning from pretrained checkpoint
+    print(f"FINETUNING ON {ckpt_filename}!")
+    futures = []
+    log_metrics_partial = partial(log_metrics, parent_ckpt, ckpt_filename, glue_metrics)
```

Comment:
nit: prefer lambda over functools.partial for pyright support.

Reply:
switched!